blog/2018-07-17-UnitTestsInXslxFileForConferenceScheduling.adoc

```diff
@@ -0,0 +1,122 @@
+= Unit tests in .xlsx files
+:page-interpolate: true
+:awestruct-author: MusaTalluzi
+:awestruct-layout: blogPostBase
+:awestruct-tags: [useCase]
+
+After you implement a planning problem, you must write unit tests for every score rule.
```

Comment:
The first paragraph of the blog is considered "the abstract". It's copy-pasted on summary pages such as
  https://www.optaplanner.org/blog/

Ideally, that first paragraph is 3-4 sentences long and explains why someone should take the time to read this article. It's far more important than the rest of the article (except the title), because:
- if the title is boring, people won't click on the link
- if the first paragraph is talking about something useful to them, people won't read the article

So, let's change this into something like:

"The business experts explain the business constraints to us, the developers. But how can we be sure that we understood them correctly? Or worse, how can we test that they agree among themselves once the constraints are formalized?

Well, there's one great way to do that: JUnit tests populated by XLS data. We allow them to recreate a small subset of the solution in Excel/LibreOffice and let them decide how many constraints match. Then our JUnit tests test if our constraint implementations adhere to those requirements."